pandas/io/excel.py

```diff
@@ -329,29 +329,38 @@ def _parse_cell(cell_contents, cell_typ):
                appropriate object"""
 
             if cell_typ == XL_CELL_DATE:
-                if xlrd_0_9_3:
-                    # Use the newer xlrd datetime handling.
-                    cell_contents = xldate.xldate_as_datetime(cell_contents,
-                                                              epoch1904)
-
-                    # Excel doesn't distinguish between dates and time,
-                    # so we treat dates on the epoch as times only.
-                    # Also, Excel supports 1900 and 1904 epochs.
-                    year = (cell_contents.timetuple())[0:3]
-                    if ((not epoch1904 and year == (1899, 12, 31)) or
-                            (epoch1904 and year == (1904, 1, 1))):
-                        cell_contents = time(cell_contents.hour,
-                                             cell_contents.minute,
-                                             cell_contents.second,
-                                             cell_contents.microsecond)
-                else:
-                    # Use the xlrd <= 0.9.2 date handling.
-                    dt = xldate.xldate_as_tuple(cell_contents, epoch1904)
 
-                    if dt[0] < MINYEAR:
-                        cell_contents = time(*dt[3:])
+                try:
+                    if xlrd_0_9_3:
+                        # Use the newer xlrd datetime handling.
+                        cell_contents = \
+                            xldate.xldate_as_datetime(cell_contents,
+                                                      epoch1904)
+
+                        # Excel doesn't distinguish between dates and time,
+                        # so we treat dates on the epoch as times only.
+                        # Also, Excel supports 1900 and 1904 epochs.
+                        year = (cell_contents.timetuple())[0:3]
+                        if ((not epoch1904 and year == (1899, 12, 31)) or
+                                (epoch1904 and year == (1904, 1, 1))):
+                            cell_contents = time(cell_contents.hour,
+                                                 cell_contents.minute,
+                                                 cell_contents.second,
+                                                 cell_contents.microsecond)
                     else:
-                        cell_contents = datetime(*dt)
+                        # Use the xlrd <= 0.9.2 date handling.
+                        dt = xldate.xldate_as_tuple(cell_contents, epoch1904)
+
+                        if dt[0] < MINYEAR:
+                            cell_contents = time(*dt[3:])
+                        else:
+                            cell_contents = datetime(*dt)
+                except OverflowError:
+                    warn("\nCell marked as date caused C int overflow while "
```

Comment:
actually this is not needed warning. Just leave the cell as it was. The result will end up being object dtype (if there are convertible dates in the column along with some non-convertible ones). Pls tests this as well.